### ISSUE TITLE
iFrame using own local storage

### ISSUE THREAD

**Issue by GiackV** [WICG/shared-storage#80]
An iFrame that uses its own localstorage, with a source that is different from the domain of the website it is contained, will be blocked by the 2024 third party cookies proposal by chrome, even if the iFrame do not read or use any content of the parent's storage.

My question is, will sharedstorage solve the problem and how? 
From what i've understood, the shared storage API (together with fenced frames) allows this particular behaviour between sites and content frames that only requires a private storage for storing data.

**Comment by richard-obrien** [comment]
Hi @menonasha,

I'm not the OP here but we also have this issue at [BugHerd](https://bugherd.com) and would love some clarity.

BugHerd is a tool that allows web developers (e.g. agencies, consultants, or in-house teams) to gather feedback from their clients & stakeholders using our embedded widget (the BugHerd Sidebar). When embedded on the customer's website, the BugHerd Sidebar will only display to users who have been invited to give feedback on the project (e.g. have a Bugherd login).

Currently, we store a user's BugHerd Login session id in a cookie which we simply use to determine on page load whether the BugHerd Sidebar should appear or not for the current website visitor. 

We have adapted our application significantly to support Firefox, Safari, and up until v113, Edge's implementation of the StorageAccessAPI, however the recent change to limit access to "first party sets" is very concerning.

We notice on the Mozilla documentation there is mention that the first party sets restriction ["...is intended to be relaxed as the implementation is developed further"](https://developer.mozilla.org/en-US/docs/Web/API/Storage_Access_API#browser_compatibility:~:text=is%20intended%20to%20be%20relaxed%20as%20the%20implementation%20is%20developed%20further.). Is there anywhere we can get more clarity on what is meant by this? Also we'd really love to know if there is someone we can talk to further about this use case as the "first party sets" direction for the storage access api can severely impact our ability to deliver our product's core functionality to customers.

Cheers,
Richard
Head of Product @ BugHerd.

**Comment by helenyc** [comment]
Hi @richard-obrien - thanks for reaching out!

Chrome is supporting the Storage Access API (SAA) for [First-Party Sets use cases](https://github.com/WICG/first-party-sets#use-cases), and also for extended use cases. You can read more about these extended use cases and Chrome's planned implementation [here](https://github.com/cfredric/chrome-storage-access-api). Please feel free to leave questions and feedback in that repo.

Thank you!

**Comment by pythagoraskitty** [comment]
Closing this issue for now. If you need further clarification about Shared Storage, please either re-open it or open a new issue.
